spring-cloud-dataflow-rest-resource/src/main/java/org/springframework/cloud/dataflow/rest/resource/DeploymentStatus.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.rest.resource;
+
+import org.springframework.util.Assert;
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+
+/**
+ * Provides a typed enumeration of deployment statuses for Streams.
+ *
+ * @author Gunnar Hillert
+ *
+ */
+@JsonFormat(shape = JsonFormat.Shape.OBJECT)
+public enum DeploymentStatus {
```

Comment:
OK, this all makes sense then. Still, could you
- add a comment somewhere explaining that the new field is transient
- change the name, as I'm really worried about term pollution here